output/full.go

```diff
@@ -26,13 +26,20 @@ import (
 func SendFull(ctx context.Context, server *state.Server, collectionOpts state.CollectionOpts, logger *util.Logger, newState state.PersistedState, diffState state.DiffState, transientState state.TransientState, collectedIntervalSecs uint32) error {
 	s := transform.StateToSnapshot(newState, diffState, transientState)
 	s.CollectedIntervalSecs = collectedIntervalSecs
-	s.CollectorErrors = logger.ErrorMessages
-
-	return submitFull(ctx, s, server, collectionOpts, logger, newState.CollectedAt, false)
+	err := verifyIntegrity(&s)
+	if err != nil {
+		logger.PrintError(fmt.Sprintf("Snapshot integrity check failed: %s; please contact support", err))
+		// Don't return an error here, since that would skip the state update, and
+		// if the integrity failure is due to a state diffing issue, we would not
+		// be able to make progress. Instead, send a failed snapshot directly.
+		return SendFailedFull(ctx, server, collectionOpts, logger)
+	} else {
+		return submitFull(ctx, s, server, collectionOpts, logger, newState.CollectedAt, false)
+	}
 }
 
 func SendFailedFull(ctx context.Context, server *state.Server, collectionOpts state.CollectionOpts, logger *util.Logger) error {
-	s := snapshot.FullSnapshot{FailedRun: true, CollectorErrors: logger.ErrorMessages}
+	s := snapshot.FullSnapshot{FailedRun: true}
```

Comment:
I ended up moving this into `submitFull` because both callers were setting collector errors independently before invoking that helper--we might as well do it in the shared code path.